combat-trainer.lic

```diff
@@ -3479,7 +3478,7 @@ class GameState
   def dance
     if @dynamic_dance_skill
       filtered_skills = weapon_training.reject { |skill, _| $non_dance_skills.include?(skill) }
-      @dance_skill = filtered_skills.min_by { |skill, _| DRSkill.getrank(skill) }.first
+      @dance_skill = sort_by_rate_then_rank(filtered_skills).first
```

Comment:
Original logic is being updated to sort by rate, too. As well as to make that decision consistent throughout combat-trainer by using game_state.sort_by_rate_then_rank. If the decision algoritm should change in the future, there's now one spot to do it.